### PR TITLE
made ContentAssistFragment2 better customizable

### PR DESCRIPTION
**Pull request by cdietrich**
made ContentAssistFragment2 better customizable
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.java [hunk 1]
```diff
@@ -241,6 +241,17 @@ protected GeneratedJavaFileAccess generateGenJavaProposalProvider() {
...
       final ArrayList<AbstractRule> remainingRules = IterableExtensions.<AbstractRule, ArrayList<AbstractRule>>fold(this.getGrammar().getRules(), CollectionLiterals.<AbstractRule>newArrayList(), _function_1);
+      _xblockexpression = this.generateGenJavaProposalProvider(assignments, remainingRules);
+    }
+    return _xblockexpression;
+  }
+  
+  /**
+   * @since 2.23
+   */
+  protected GeneratedJavaFileAccess generateGenJavaProposalProvider(final List<Assignment> assignments, final List<AbstractRule> remainingRules) {
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.generator/xtend-gen/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.java:252]: the current api here really is bad and i did not want to break. what do you think?

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend [hunk 2]
```diff
@@ -159,11 +160,25 @@ class ContentAssistFragment2 extends AbstractInheritingFragment {
 		// take the non-abstract class signature for the src-gen class in case of !generateStub
 		//  as proposalProviders of sub languages refer to 'grammar.proposalProviderClass',
 		//  see 'getGenProposalProviderSuperClass(...)'
+		generateGenJavaProposalProvider(assignments, remainingRules)
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend:163]: While we are at it: does it make sense to extract methods `getAssignments` and `getRemainingRules`?

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend [hunk 3]
```diff
@@ -159,11 +160,25 @@ class ContentAssistFragment2 extends AbstractInheritingFragment {
...
 			if (isGenerateStub) grammar.genProposalProviderClass else grammar.proposalProviderClass;
 		
+		val superClass = grammar.genProposalProviderSuperClass
+		generateGenJavaProposalProvider(assignments, remainingRules, genClass, superClass)
+	}
+
+	/**
+	 * @since 2.23
+	 */
+	protected def GeneratedJavaFileAccess generateGenJavaProposalProvider(List<Assignment> assignments, List<AbstractRule> remainingRules, TypeReference genClass, TypeReference superClass) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend:180]: I'd like to have a better name for `remainingRules`, maybe just `rules` or `rulesToComplete` ? Same for the method above.

**Review by szarnekow — APPROVED** [review]
I'm ok with the change. Maybe consider the two small suggestions from the comments, though.

**Review by szarnekow — APPROVED** [review]
:+1:

**Pull request merged**